### PR TITLE
[FEATURE REQUEST] Enforce OIDC auth flow

### PR DESCRIPTION
**Pull request by JuancaG05**
## Related Issues
App: https://github.com/owncloud/android/issues/4500

- [x] Add changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
- [x] Add feature to Release Notes in `ReleaseNotesViewModel.kt` creating a new `ReleaseNote()` with String resources (if required)

_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/authentication/oauth/OAuthUtils.kt [hunk 1]
```diff
@@ -124,9 +124,10 @@ class OAuthUtils {
...
         fun buildRedirectUri(context: Context): Uri =
-            Uri.Builder()
-                .scheme(context.getString(R.string.oauth2_redirect_uri_scheme))
-                .authority(context.getString(R.string.oauth2_redirect_uri_path))
-                .build()
+            Uri.Builder().apply {
+                scheme(context.getString(R.string.oauth2_redirect_uri_scheme))
+                authority(context.getString(R.string.oauth2_redirect_uri_host))
+                path(context.getString(R.string.oauth2_redirect_uri_path))
+            }.build()
```
Comment (joragua) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/authentication/oauth/OAuthUtils.kt:131]: I think `apply` is not necessary in this case

**Review by joragua — CHANGES_REQUESTED** [review]
Only one comment here! 💪🏻 @JuancaG05

File: owncloudApp/src/main/java/com/owncloud/android/presentation/authentication/oauth/OAuthUtils.kt (hunk 1 above)
Comment (JuancaG05) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/authentication/oauth/OAuthUtils.kt:131]: Changed!

**Review by joragua — APPROVED** [review]
LGTM! Let's move it to QA 🚀

**Comment by jesmrec** [comment]
QA checks:

`enforce_oidc` is false

- [x] Regular oC server -> correct authentication & authorization
- [X] Server with only OIDC -> error (shows basic auth credential fields)
- [x] Basic auth -> correct authentication & authorization
- [x] OAuth2 ->  correct authentication & authorization

`enforce_oidc` is true

- [ ] Regular oC server -> error
- [ ] Server with only OIDC -> correct authentication & authorization
- [ ] Basic auth -> error
- [ ] OAuth2 -> error

- [ ] `oauth2_redirect_uri_path` is empty
- [ ] `oauth2_redirect_uri_path` is not empty

**Comment by jesmrec** [comment]
Great job!! problem solved with a clean solution 💯 

Ready to go

**Pull request merged**